src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
     /// <summary>
     /// Factory for <see cref="TagHelperDescriptor"/>s from <see cref="ITypeInfo"/>s.
     /// </summary>
-    public static class TagHelperDescriptorFactory
+    public class TagHelperDescriptorFactory
```

Comment:
Understand `TagHelperDescriptorFactory` doesn't itself hit the disk. But it's not clear caching at the lower `TagHelperDesignTimeDescriptorFactory` level is the way to do. This class is rather allocation-heavy and could contain the cached information.